src/Illuminate/Routing/Matching/SubdomainValidator.php

```diff
@@ -0,0 +1,28 @@
+<?php
+
+namespace Illuminate\Routing\Matching;
+
+use Illuminate\Http\Request;
+use Illuminate\Routing\Route;
+
+class SubdomainValidator implements ValidatorInterface
+{
+    /**
+     * Validate a given rule against a route and request.
+     *
+     * @param  \Illuminate\Routing\Route  $route
+     * @param  \Illuminate\Http\Request  $request
+     * @return bool
+     */
+    public function matches(Route $route, Request $request)
+    {
+        if (is_null($route->subdomain())) {
+            return true;
+        }
+
+        $subdomains = implode('|', $route->subdomain());
+        $regex = "/^$subdomains\..*/i";
```

Comment:
For the regex syntax, shouldn't there be parentheses around the group of `|`-delimited entries?

ie: 

```diff
-$regex = "/^$subdomains\..*/i";
+$regex = "/^($subdomains)\..*/i";
```

Otherwise the `|` could get processed incorrectly ... and also the `\.` that's supposed to come before the TLD will only come after the _last_ subdomain in the list.